### PR TITLE
added ga-bank homework, and updated cat homework

### PR DESCRIPTION
**Pull request by j-tws**
_If you had no problems you can just leave it empty_

- How difficult did you find this (out of 10)? (0 being no problems at all, 10 being impossible)
7/10

- Was there anything that you struggled with?
struggled abit in making some code working, but overall managed to get the whole bank working. Extra thanks for helping me wtih the homework Kris

- Is there any particular code you want me to look into?
Probably next step is DRY the code. Already got a few feedback from the homework demo but would still like to see if there are any ways to make it DRYer

- Is there anything that you'd like some further information on?
- Roughly how long did it take?
- 6-8 hours

### CODE REVIEW

**Comment by kristabel-wong** [comment]
Hi Justin!

Good work on the homework! But it only works if you follow the correct process i.e. making sure to enter positive values > deposit/withdraw. What happens if i enter something different? You probably did get some of the below in the hw demo but I was not across it cause i was marking homework. See below:

**Somethings to consider:**
- pressing deposit with no value entered returns a NaN
- Depositing negative values ($-100, balance 0) will negatively affect the balance. the amount will be removed, rather than added to balance
- Likewise withdrawing negative values will add to the balance rather than withdraw from the balance
- entering in invalid characters like letters will also render a NaN in the balance

**Further:**
- when withdrawing from the 2 accounts, will both turn red if they have $0 balance? (only one turns red atm)

For ways to make it DRY-er I will try make suggestions in the code review for you

File: justin-tan/week2/fri/ga-bank/js/atm.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    //when you have insufficient total balance to withdraw
+    } else if ( $savingsAmount > totalBalance ) {
+        $('#savings-balance').text(`Insufficient $$`);
+    }
+}
+
+//===========================================================
+
+//function for SAVINGS BALANCE background color
+const backgroundSavingsColor = function () {
```
Comment (kristabel-wong) [inline justin-tan/week2/fri/ga-bank/js/atm.js:175]: you already have a `backgroundCheckingColor` that does the same thing
you can refactor that to take in a parameter, being the account balance name

```
const backgroundBalance = function (accBalance) { // `#checking-balance` or  #savings-balance'
    if ( ($checkingBalance === 0 || $(accBalance).text() === `Insufficient $$`) || ($savingBalance === 0)) {
        $(accBalance).css('background-color', 'red');
    } else {
        $(accBalance).css('background-color', 'rgb(227, 227, 227');
    }
}


$('#checking-withdraw').on( 'click', backgroundBalance('#checking-balance');


```

File: justin-tan/week2/fri/ga-bank/js/atm.js [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+//============================================================
+
+//event handlers for savings account deposit and withdrawal buttons
+$('#savings-deposit').on( 'click', savingsDeposit );
+$('#savings-withdraw').on( 'click', savingsWithdrawal );
+
+//event handlers for checking account background color
+$('#savings-withdraw').on( 'click', backgroundSavingsColor );
+$('#savings-deposit').on( 'click', backgroundSavingsColor );
```
Comment (kristabel-wong) [inline justin-tan/week2/fri/ga-bank/js/atm.js:191]: you could potentially put the background colour function inside the withdraw function. its just another if statement or additional statement inside an existing if statement, then you add the css line within that

**Comment by j-tws** [comment]
Kris thanks for the feedback. I've checked my bank hw and both accounts should show red background when amount is 0 🤔 

Apart from that thanks for the code review as well. I'll look into DRYing my code and take consideration points you've listed when I revisit the ga-bank!

**Pull request merged**